lib/pool.js

```diff
@@ -3,93 +3,95 @@ var EventEmitter = require('events').EventEmitter;
 var defaults = require('./defaults');
 var genericPool = require('generic-pool');
 
-var pools = {
-  //dictionary of all key:pool pairs
-  all: {},
-  //reference to the client constructor - can override in tests or for require('pg').native
-  Client: require('./client'),
```

Comment:
I was 'punching over' this which is bad since this is a singleton.  I made it not a singleton since there's no reason for it to be one, and I pass the proper constructor (native or JS) when its initialized.